java-sdk/sdk-checkpoints.asciidoc

```diff
@@ -0,0 +1,88 @@
+= Checkpoints in the Connectors SDK
+
+== Use Cases
+
+=== Incremental Re-crawl
+
+Incremental re-crawl can be supported when a Changes API is available (e.g., Jive, Salesforce, OneDrive). When a Changes API is available, it’s necessary to provide an input parameter to be tracked, such as a date, link, page token, or other. The input parameter is generated (retrieved) while running the first job. During the next job, that parameter will be used to query the Changes API and retrieve new, modified, and deleted objects.
+
+The SDK provides a way to store the input parameters, establishing *checkpoints*, and use them in the subsequent jobs.
+
+[[checkpoint-design]]
+==== Checkpoint Design
+
+Fetcher implementations can emit checkpoint messages by calling:
+
+```
+fetchContext.emitCheckpoint(CHECKPOINT)_ID, checkpointMetadata);
+```
+
+After the checkpoint is emitted, Fusion will handle this message as follows:
+
+* The checkpoint will not be used in the current job.
+* The checkpoint will be stored in the CrawlDB with the appropriate status.
+
+In subsequent job runs, Fusion will check the CrawlDB for any previously stored checkpoints. If any are available, only those checkpoints will be sent to the fetchers; no other input types will be sent. If checkpoints are not available, all other items in the CrawlDB (Documents, FetchInputs, Errors, etc.) will be sent to the fetchers instead.
+
+IMPORTANT:  In order to update a checkpoint, it must be emitted using its original ID. The ID is the only way the SDK controller can identify and update a checkpoint.
+
+===== First Job Flow
+
+image:/assets/images/sdkcheck-1stflow.png[First Job Flow]
```

Comment:
Can we link the step number to an specific arrow in the image(like `sdkcheck-itemcomplete.png`)? It's not clear what the flow is.